services/elastic_ip_manager/src/main/java/com/futurewei/alcor/elasticipmanager/service/implement/ElasticIpServiceImpl.java

```diff
@@ -86,9 +93,11 @@ public ElasticIpInfo createElasticIp(ElasticIpInfo request) throws Exception {
         eip.setElasticIp(ipAddress);
```

Comment:
Line 79-89 in the new file (I am unable to comment there directly):

In the current algorithm, we look into all ranges and pick the first one matching the request ip's version. Pulling all ranges from repo is fine when the number of ranges is limited.

One concern is that the first few ranges could be drained very quickly, which causes the following requests takes more time to find one available ip as it will need to check more and more ranges. We could think of adding ip availability stats for each range, and pick the one with least utilization.

Not high priority. I would recommend us to run some perf/scalability test to see. @Eric-Yuan